python/ray/tune/schedulers/pbt.py

```diff
@@ -821,17 +865,17 @@ def on_trial_result(self, trial_runner: "trial_runner.TrialRunner",
         if reset_successful:
             trial_executor.restore(trial, checkpoint, block=True)
         else:
-            trial_executor.stop_trial(
-                trial, destroy_pg_if_cannot_replace=False)
+            trial_executor.pause_trial(trial)
             trial.set_experiment_tag(new_tag)
             trial.set_config(new_config)
-            trial_executor.start_trial(trial, checkpoint, train=False)
+            trial.on_checkpoint(checkpoint)
```

Comment:
Why on checkpoint?

Reply:
That's a great question!
Without this statement, the post-transfer trial will just start from fresh by outer Tune loop instead of being started from a previous checkpoint of the "transferred over" trial. 
This is an equivalent of `checkpoint` argument [here](https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/tune/schedulers/pbt.py?L828).